include/signal.h

```diff
@@ -497,13 +506,16 @@ int  raise(int signo);
 int  sigaction(int signo, FAR const struct sigaction *act,
                FAR struct sigaction *oact);
 int  sigaddset(FAR sigset_t *set, int signo);
+int  sigandset(FAR sigset_t *dest, FAR sigset_t *left, FAR sigset_t *right);
```

Comment:
should we move the new api to nuttx/signal.h and add nx

Reply:
Let's have all interfaces:
```
int sigandset(FAR sigset_t *dest, FAR sigset_t *left, FAR sigset_t *right);
int signandset(FAR sigset_t *dest, FAR sigset_t *left, FAR sigset_t *right);
int sigorset(FAR sigset_t *dest, FAR sigset_t *left, FAR sigset_t *right);
int sigxorset(FAR sigset_t *dest, FAR sigset_t *left, FAR sigset_t *right);
```
in libc.

Note:
- `nxsig_delmultiple` -> `signandset`
- `nxsig_xorset` -> `sigxorset`

Reply:
sigisemptyset, sigorset, and sigandset, are semi-standard APIs that are defined in signal.h (if _GNU_SOURCE is defined).  https://linux.die.net/man/3/sigandset

I think that they are in the right place.

There is no such semi-standard for nxsig_delmultiple or nxsig_xorset.  I don't know how you see this, but I think we should avoid exposing interfaces that have no basis in any standard.  Putting them in include/nuttx/signal.h is what we do if they are non-standard and we don't want them to corrupt the integrity of the standard OS interface.

If they are in include/nuttx/signal.h, should the nxsig_ prefix be retained?  They are not recommended for use by applications because they are non-standard and non-portable.

something_nandset is a better name.

What do you guys think?

Reply:
I do not mind

Reply:
Yes, this is a good seperation.